File: sdks/dotnet/src/Models/Stocktake.cs
using System;
using Newtonsoft.Json;

namespace Puxbay.SDK.Models
{
    public class Stocktake
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("branch")]
        public string Branch { get; set; }

        [JsonProperty("status")]
        public string Status { get; set; }

        [JsonProperty("notes")]
        public string Notes { get; set; }

        [JsonProperty("created_at")]
        public DateTime? CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public DateTime? UpdatedAt { get; set; }
    }
}

File: sdks/php/src/Models/Stocktake.php
<?php

namespace Puxbay\Models;

class Stocktake
{
    public function __construct(
        public readonly string $id,
        public readonly string $branch,
        public readonly string $status,
        public readonly ?string $notes = null,
        public readonly \DateTimeImmutable $createdAt,
        public readonly \DateTimeImmutable $updatedAt,
    ) {
    }

    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'],
            branch: $data['branch'],
            status: $data['status'],
            notes: $data['notes'] ?? null,
            createdAt: new \DateTimeImmutable($data['created_at']),
            updatedAt: new \DateTimeImmutable($data['updated_at']),
        );
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'branch' => $this->branch,
            'status' => $this->status,
            'notes' => $this->notes,
        ];
    }
}
